contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -1070,15 +1070,63 @@ validate_special_function(char *func_nsname, char *func_name, int nargs, Oid *in
 
 	/* If function is not a special function no additional handling required */
 	if (special_func == NULL)
+	{
+		/* report error for case when NULL casted to different datatypes and passed as 2nd or 3rd argument of SUBSTRING() function */
+		if (strlen(func_name) == 9 && strncmp(func_name, "substring", 9) == 0)
+		{
+			for (int i = 1; i < nargs; i++)
+			{
+				if (input_typeids[i] != UNKNOWNOID)
```

Comment:
NIT: 
instead of
```
if (input_typeids[i] != UNKNOWNOID)
{
 do something
}

```
you can do 
```
if (input_typeids[i] == UNKNOWNOID)
      continue;

do something

```

Reply:
Done